lib/datadog/appsec/processor.rb

```diff
@@ -0,0 +1,113 @@
+# typed: ignore
+
+require 'datadog/appsec/assets'
+
+module Datadog
+  module AppSec
+    # Processor integrates libddwaf into datadog/appsec
+    class Processor
+      # Interface object to check using case .. when
+      module IOLike
+        def read; end
+        def rewind; end
+
+        def self.===(other)
+          instance_methods.reduce(true) { |ok, meth| ok && other.respond_to?(meth) }
+        end
+      end
+
+      def initialize
+        unless load_libddwaf && load_ruleset && create_waf_handle
+          Datadog.logger.warn { 'appsec is disabled, cause: errors' }
+        end
+      end
+
+      def ready?
+        !@ruleset.nil? && !@handle.nil?
+      end
+
+      def context
+        Datadog::AppSec::WAF::Context.new(@handle)
+      end
+
+      private
+
+      def load_libddwaf
+        Processor.require_libddwaf && Processor.libddwaf_required?
+      end
+
+      def load_ruleset
+        ruleset_setting = Datadog::AppSec.settings.ruleset
+
+        begin
+          @ruleset = case ruleset_setting
+                     when :recommended, :risky, :strict
+                       JSON.parse(Datadog::AppSec::Assets.waf_rules(ruleset_setting))
+                     when String
+                       JSON.parse(File.read(ruleset_setting))
+                     when IOLike
+                       JSON.parse(ruleset_setting.read).tap { ruleset_setting.rewind }
+                     when Hash
+                       ruleset_setting
+                     else
+                       raise ArgumentError, "unsupported value for ruleset setting: #{ruleset_setting.inspect}"
+                     end
```

Comment:
I don't consider this a blocker for this PR, but I would like to try and push back on the amount of flexibility in configuring this. 

It looks to me that only supporting both symbols + string with a path should cover most (all?) use cases.

Do we need all of this flexibility? If it's just a niche thing for a small amount of customers, is it worth the extra burden to support? How often do we expect users to use "IOLike" and "Hash-based" rulesets in production?

Reply:
> How often

With remote ruleset updates via remote config coming up, much.

I understand the push back, but from experience this is a great tool to have when debugging and in support scenarios.

Reply:
Fair enough, thanks for adding the extra context 👍